### DISCUSSION TITLE
Crop "double pages" mangas on simple pages

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#4555] [Help] [answered]
Hello everyone, 

I tried to crop "double pages" mangas in  simple pages with this formula:  mogrify -crop 2x1@ -format jpg *.

The double pages are well cut to their center in two separate files, a file on the left page named "Filename-0" and a file of the right page named "Filename-1".

For books read from left to right this is not a problem, but for mangas lying from right to the left that poses a problem because the order of the cutting files is reversed.

Would anyone have an idea of formula that can switch the names "-0" and "-1" of outgoing files?

I try to find something, including the use of the "-flop" function, but without success. 

Thanks in advance

**Comment by fmw42** [comment]
Flop the images before and after

`mogrify -format jpg -flop -crop "2x1@" -flop *`

**Reply by ghost** [reply]
Hi fmw42, thank you for your answer.

I have already tried the formula you suggest and the problem is that the -0 file which comes out inverted, like a mirror effect.

**Comment by fmw42** [comment]
I do not think mogrify will work that way. It was not intended to do cropping. You should write a script loop over your files and do the same. For example in Unix syntax, this works with convert.  I cd to the folder with the images. Do the loop processing with convert and write the results to an existing folder to hold the two output images per input image.
Convert can handle the crop and the flopping of both outputs properly.

```
cd
cd desktop/test1
for img in *.jpg; do
convert $img -flop -crop "2x1@" -flop ../test2/$img
done

```

Sorry, I do not know the looping structure for Windows .bat files. Perhaps one of the other users can convert my command for you.

**Reply by ghost** [reply]
Thank you very much for your detailed answer;).

**Comment by snibgo** [comment, marked as answer]
> Would anyone have an idea of formula that can switch the names "-0" and "-1" of outgoing files?

Insert "-reverse" or "+swap" before the input/output filename. Beware that the input file will be replaced by both the output files.

**Reply by ghost** [reply]
Hi snibgo,

Thank you very much for your answer, it works!

Congratulations on your deduction.

The input file in my case does not erase but has a "~" at the end of his extension after the operation. 

Have a good day.
